10/src/javascripts/app.js

```diff
@@ -0,0 +1,25 @@
+let wd = require('./wd.js');
+
+window.onload = function () {
+    wd.createRoot('app')
+        .routes([
+            { url: '/', templateUrl: 'routes/home.html' },
+            { url: '/sign-in', templateUrl: 'routes/sign-in.html' }
+        ])
+        .component("RouterLink", {
+            template: "<a href=\"\"><\/a>",
+            beforeMount: function (app, element, componentData) {
+                //console.log("beforeMount", app, element, componentData);
+                let replacement = document.createElement("a");
+                replacement.href = element.getAttribute("href");
```

Comment:
Here you've set `href` property for the `replacement` link, and it works as expected. But in other cases there can be a problem because of difference between of DOM element attribute and property of element object. To avoid errors, please use `setAttribute` for setting attributes.